src/events_watcher/substrate/signature_bridge_watcher.rs

```diff
@@ -279,6 +309,11 @@ where
     }
 }
 
+pub fn parse_nonce_from_proposal_data(proposal_data: &[u8]) -> u32 {
+    let nonce_bytes = proposal_data[36..40].try_into().unwrap_or_default();
+    u32::from_be_bytes(nonce_bytes)
+}
```

Comment:
I'm sure parsing the full proposal header here is safer, and even more readable:
```suggestion
pub fn parse_nonce_from_proposal_data(proposal_data: &[u8]) -> u32 {
	assert!(proposal_data.len() >= ProposalHeader::LENGTH);
	let mut header_bytes = [0u8; ProposalHeader::LENGTH];
	header_bytes.copy_from_slice(&proposal_data[0..ProposalHeader::LENGTH]);
	let header = ProposalHeader::from(header_bytes);
    header.nonce.to_u32()
}
```